### PR TITLE
add user defined function support in quantization and unittest and demo

### PR DESCRIPTION
**Pull request by slf12**
add user defined function support in quantization and unittest and demo
对应pr  https://github.com/PaddlePaddle/Paddle/pull/24720
image

### CODE REVIEW

File: demo/quant/pact_quant_aware/README.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+2020-06-05 15:26:08,598-INFO: epoch[0]-batch[30] - loss: 1.427333951; acc_top1: 0.6953125; acc_top5: 0.875; time: 1.51066279411
+2020-06-05 15:26:24,009-INFO: epoch[0]-batch[40] - loss: 1.43955898285; acc_top1: 0.6640625; acc_top5: 0.8671875; time: 1.49221611023
+2020-06-05 15:26:39,501-INFO: epoch[0]-batch[50] - loss: 1.29342699051; acc_top1: 0.6953125; acc_top5: 0.90625; time: 1.50851297379
+2020-06-05 15:26:54,927-INFO: epoch[0]-batch[60] - loss: 1.49478590488; acc_top1: 0.6171875; acc_top5: 0.875; time: 1.50131177902
+2020-06-05 15:27:10,250-INFO: epoch[0]-batch[70] - loss: 1.34970903397; acc_top1: 0.7109375; acc_top5: 0.890625; time: 1.51333618164
+2020-06-05 15:27:25,309-INFO: epoch[0]-batch[80] - loss: 1.51600492001; acc_top1: 0.6796875; acc_top5: 0.859375; time: 1.44952607155
+2020-06-05 15:27:40,273-INFO: epoch[0]-batch[90] - loss: 1.5926772356; acc_top1: 0.6328125; acc_top5: 0.859375; time: 1.45620679855
+2020-06-05 15:27:55,660-INFO: epoch[0]-batch[100] - loss: 1.40280032158; acc_top1: 0.671875; acc_top5: 0.875; time: 1.50846099854
+```
+可以看出loss值比较稳定，并且我们在实验时，可以得到top-1 77.6%的量化模型。
```
Comment (qingqing01) [inline demo/quant/pact_quant_aware/README.md]: 这里说下实验条件，以及给下最终量化之后的模型

File: demo/quant/pact_quant_aware/train.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+def create_optimizer(args):
+    if args.lr_strategy == "piecewise_decay":
+        return piecewise_decay(args)
+    elif args.lr_strategy == "cosine_decay":
+        return cosine_decay(args)
+
+
+def compress(args):
+    ############################################################################################################
```
Comment (qingqing01) [inline demo/quant/pact_quant_aware/train.py]: 不超过80个字符

File: paddleslim/quant/quanter.py [hunk 3]
```diff
@@ -175,7 +185,32 @@ def quant_aware(program, place, config=None, scope=None, for_test=False):
             `fluid.global_scope <https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api_cn/executor_cn/global_scope_cn.html>`_.              When ``None`` will use `fluid.global_scope() <https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api_cn/executor_cn/global_scope_cn.html>`_ . Default: ``None``.
         for_test(bool): If the 'program' parameter is a test program, this parameter should be set to ``True``. 
             Otherwise, set to ``False``.Default: False
-    
+       weight_quantize_func(function): Function that defines how to quantize weight. Using this
```
Comment (qingqing01) [inline paddleslim/quant/quanter.py:188]: 同样补下中文API文档吧

File: demo/quant/pact_quant_aware/README.md (hunk 1 above)
Comment (slf12) [reply demo/quant/pact_quant_aware/README.md]: done

File: demo/quant/pact_quant_aware/train.py (hunk 2 above)
Comment (slf12) [reply demo/quant/pact_quant_aware/train.py]: done

File: paddleslim/quant/quanter.py (hunk 3 above)
Comment (slf12) [reply paddleslim/quant/quanter.py:188]: done

**Review by wanghaoshuang — APPROVED** [review]

**Pull request merged**